runtime/util/vmargs.c

```diff
@@ -1355,8 +1388,13 @@ addEnvironmentVariableArguments(J9PortLibrary* portLib, const char* envVarName,
 	}
 	JVMINIT_VERBOSE_INIT_TRACE1(verboseFlags, "Parsing environment variable %s\n", envVarName);
 	j9sysinfo_get_env(envVarName, argumentBuffer, envVarSize);
-
-	return parseOptionsBuffer(portLib, argumentBuffer, vmArgumentsList, verboseFlags, TRUE);
+	UDATA envVarNameLength = LITERAL_STRLEN(envVarName);
+	if ((J9UTF8_DATA_EQUALS(ENVVAR_JAVA_TOOL_OPTIONS, LITERAL_STRLEN(ENVVAR_JAVA_TOOL_OPTIONS), envVarName, envVarNameLength)) || (J9UTF8_DATA_EQUALS(ENVVAR_OPENJ9_JAVA_OPTIONS, LITERAL_STRLEN(ENVVAR_OPENJ9_JAVA_OPTIONS), envVarName, envVarNameLength)) || (J9UTF8_DATA_EQUALS(ENVVAR_IBM_JAVA_OPTIONS, LITERAL_STRLEN(ENVVAR_IBM_JAVA_OPTIONS), envVarName, envVarNameLength))) {
```

Comment:
Nitpick: when you have a large conditional statement like this put the `||` in a new line 

ie.

```
if (condition1
     || condition2
     ...
     || conditionN
) {
```